rust/processor/src/models/token_models/ans_lookup.rs

```diff
@@ -74,107 +77,225 @@ impl OptionalString {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NameRecordKeyV1 {
+    domain_name: String,
+    subdomain_name: OptionalString,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NameRecordV1 {
+    #[serde(deserialize_with = "deserialize_from_string")]
+    expiration_time_sec: BigDecimal,
+    #[serde(deserialize_with = "deserialize_from_string")]
+    property_version: BigDecimal,
+    target_address: OptionalString,
+}
+
 impl CurrentAnsLookup {
     pub fn from_transaction(
         transaction: &TransactionPB,
-        ans_contract_address: Option<String>,
+        maybe_ans_primary_names_table_handle: Option<String>,
+        maybe_ans_name_records_table_handle: Option<String>,
     ) -> HashMap<CurrentAnsLookupPK, Self> {
         let mut current_ans_lookups: HashMap<CurrentAnsLookupPK, Self> = HashMap::new();
-        if let Some(addr) = ans_contract_address {
-            // Extracts events and user request from genesis and user transactions. Other transactions won't have coin events
-            let txn_data = transaction
-                .txn_data
+        let txn_version = transaction.version as i64;
+        let txn_data = transaction
+            .txn_data
+            .as_ref()
+            .expect("Txn Data doesn't exit!");
+
+        // Extracts from user transactions. Other transactions won't have any ANS changes
+        if let (
+            TxnData::User(_),
+            Some(ans_primary_names_table_handle),
+            Some(ans_name_records_table_handle),
+        ) = (
+            txn_data,
+            maybe_ans_primary_names_table_handle,
+            maybe_ans_name_records_table_handle,
```

Comment:
oh this is interesting I didnt' realize you could do that lol.

Reply:
This is combining a bunch of `if lets` into one, I didn't know we could do that either.